src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala

```diff
@@ -5,24 +5,33 @@ import org.scalatest.FunSuite
 
 class InferSchemaSuite extends FunSuite {
 
-  test("String fields types are inferred correctly from null types"){
+  test("String fields types are inferred correctly from null types") {
     assert(InferSchema.inferField(NullType, "") == NullType)
     assert(InferSchema.inferField(NullType, null) == NullType)
     assert(InferSchema.inferField(NullType, "100000000000") == LongType)
     assert(InferSchema.inferField(NullType, "60") == IntegerType)
     assert(InferSchema.inferField(NullType, "3.5") == DoubleType)
     assert(InferSchema.inferField(NullType, "test") == StringType)
+    assert(InferSchema.inferField(NullType, "2015-08-20 15:57:00") == TimestampType)
   }
 
-  test("String fields types are inferred correctly from other types"){
+  test("String fields types are inferred correctly from other types") {
     assert(InferSchema.inferField(LongType, "1.0") == DoubleType)
     assert(InferSchema.inferField(LongType, "test") == StringType)
     assert(InferSchema.inferField(IntegerType, "1.0") == DoubleType)
     assert(InferSchema.inferField(DoubleType, null) == DoubleType)
     assert(InferSchema.inferField(DoubleType, "test") == StringType)
+    assert(InferSchema.inferField(LongType, "2015-08-20 14:57:00") == TimestampType)
+    assert(InferSchema.inferField(DoubleType, "2015-08-20 15:57:00") == TimestampType)
   }
 
-  test("Type arrays are merged to highest common type"){
+  test("Timestamp field types are inferred correctly from other types") {
+    assert(InferSchema.inferField(IntegerType, "2015-08-20 14") == StringType)
+    assert(InferSchema.inferField(DoubleType, "2015-08-20 14:10") == StringType)
+    assert(InferSchema.inferField(LongType, "2015-08 14:49:00") == StringType)
```

Comment:
These malformed timestamps (for example they are missing minutes, seconds or dates).